src/main/java/snownee/lychee/recipes/ShapedCraftingRecipe.java

```diff
@@ -127,11 +132,11 @@ public boolean matches(CraftingInput input, Level level) {
 		outer:
 		for (matchX = 0; matchX <= input.width() - getWidth(); ++matchX) {
 			for (matchY = 0; matchY <= input.height() - getHeight(); ++matchY) {
-				if (pattern.callMatches(CraftingInput.of(matchX, matchY, input.items()), true)) {
+				if (pattern.callMatches(input, true)) {
```

Comment:
这里是需要读取到配方在容器内的相对位置，这样修改之后 matchX 和 matchY 完全忽略，和原有逻辑不一致，我觉得应该逻辑如下
```suggestion
				if (pattern.callMatches(CraftingInput.of(input.width() - matchX, input.height() - matchY, input.items()), true)) {
```

Reply:
![动画](https://github.com/user-attachments/assets/4c7f5c54-a393-4ba4-afd7-b07fdac23db4)

直接判断的话，没问题，修改后重复触发导致第一格的物品被消耗

Reply:
这里应该是允许2*2可以偏移合成？这样修改应该是不可以的，不过我不确定

Reply:
![动画](https://github.com/user-attachments/assets/8efb78f1-3f4d-43c5-950e-5e52fa93f8a9)
![1](https://github.com/user-attachments/assets/782e4ef6-955c-4d9e-a4a2-f097590abc88)
好像没问题...
修改后反而会吞物品

Reply:
那这里 matchX 和 matchY 就无意义了，

Reply:
hm，下面有使用 https://github.com/Snownee/Lychee/blob/1.21-fabric/src/main/java/snownee/lychee/recipes/ShapedCraftingRecipe.java#L211

Reply:
我看了下是为了在合成后对不消耗的物品进行放回，这里 matchX 和 Y 还是需要

Reply:
emmm,如果去除这里两个for，只保留判断的代码的话，则会恢复到最开始的无法匹配的问题。所以只修改判断的部分，matchX和Y还是会被增加，应该不会影响下面的使用

Reply:
对比了一下，matchX和Y似乎是距离最小能够合成的配方的距离
![image](https://github.com/user-attachments/assets/5ec69dcc-2b2d-498e-a5a4-27e162bf71ef)
![image](https://github.com/user-attachments/assets/6911ea64-5a30-4f88-8ae7-b71d609d3417)
但是好像不会影响到下面的使用（？
![image](https://github.com/user-attachments/assets/29819687-97c6-4b70-b1cc-b290accdbda0)